### DISCUSSION TITLE
Postgres works from localhost but not in production

### DISCUSSION THREAD

**Discussion by raphapizzi** [vercel/community#2309] [Help]
### Summary

Hello, I have a project running with Astro.

I created a Storage of the Postgres type and managed to make it work perfectly from my localhost, I make requests and I can write/read in the remote Postgres.

But when I deploy to Production Postgres doesn't work, at the same time I don't get any errors either.

The environment keys are correct. I also manually entered key by key and even entered them manually and it didn't work.

I have other environment variables that are not Postgres and they work normally (discarding some problems with the variables).

Can anyone help me?

### Additional information

When I run the command `vercel dev` on my local machine, I get this output:
```
Vercel CLI 29.1.1
> Running Dev Command “astro dev --port $PORT”
  🚀  astro  v2.4.1 started in 493ms
  
  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose
  
> Ready! Available at http://localhost:3000
Message saved to database
```
That is, I can run locally and I can save in remote Postgres without any problem.

But if I run `npm run dev` on my local machine I get the following output when I try to save something to the database:
```
npm run dev         

> project@0.0.1 dev
> astro dev

  🚀  astro  v2.4.1 started in 599ms
  
  ┃ Local    http://localhost:3000/
  ┃ Network  use --host to expose

[VercelPostgresError]: VercelPostgresError - 'missing_connection_string': You did not supply a 'connectionString' and no 'POSTGRES_URL' env var was found.
```
That is, using npm the database stops working, it simply does not find the variables in the .env

To deploy to Vercel in any way I do, either by committing via Github or using the `vercel deploy` or just `vercel` command, it uploads the version normally to production, but the database does not work and I also do not receive any errors in the logs.

**Comment by Paperized** [comment]
Up, did you find a solution? I'm stuck with this aswell and its been 2 days
